resteasy-jaxrs/src/main/java/org/jboss/resteasy/spi/ResteasyProviderFactory.java

```diff
@@ -2010,13 +2010,30 @@ protected void processProviderContracts(Class provider, Integer priorityOverride
       }
       if (isA(provider, Feature.class, contracts))
       {
+         ConstrainedTo constrainedTo = (ConstrainedTo) provider.getAnnotation(ConstrainedTo.class);
          int priority = getPriority(priorityOverride, contracts, Feature.class, provider);
          Feature feature = injectedInstance((Class<? extends Feature>) provider);
-         if (feature.configure(new FeatureContextDelegate(this)))
-         {
-            enabledFeatures.add(feature);
+         if (constrainedTo != null && constrainedTo.value() == RuntimeType.SERVER && getRuntimeType() == RuntimeType.SERVER) {

```

Comment:
How about replacing this and the following if clauses with:
if (contrainedTo == null || getRuntimeType() == constrainedTo.value())

Reply:
Yep, I originally thought about it, but I wanted to keep the same style as for dynamic features. Will change it then. Thanks!